server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotShardsStats.java

```diff
@@ -128,4 +141,87 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
+        int totalShards = 0;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                String currentName = parser.currentName();
+                if (currentName.equals(Fields.INITIALIZING)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    initializingShards = parser.intValue();
+                } else if (currentName.equals(Fields.STARTED)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    startedShards = parser.intValue();
+                } else if (currentName.equals(Fields.FINALIZING)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    finalizingShards = parser.intValue();
+                } else if (currentName.equals(Fields.DONE)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    doneShards = parser.intValue();
+                } else if (currentName.equals(Fields.FAILED)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    failedShards = parser.intValue();
+                } else if (currentName.equals(Fields.TOTAL)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    totalShards = parser.intValue();
+                } else {
+                    throw new ElasticsearchParseException("failed to parse snapshot shards stats, unexpected field [{}]", currentName);
+                }
+            } else {
+                throw new ElasticsearchParseException("failed to parse snapshot shards stats");
```

Comment:
this is making the parser strict and I think it breaks forward compatibility. What if one day we add a new array or object at this level? This is also valid for the other parsers added with this PR. Converting them to object parser and providing true as `supportsUnknownFields` argument is also a valid fix.